src/utils.ts

```diff
@@ -246,15 +246,42 @@ export function unixTime(): number {
   return Math.floor(Date.now() / 1000);
 }
 
-export const formatTimestamp = (t: number): string => {
-  return new Date(t * 1000).toLocaleString("en-EN", { dateStyle: "full", timeStyle: "long" });
+export const formatTimestamp = (timestamp: number, current = new Date().getTime()): string => {
+  const units: Record<string, number> = {
+    year: 24 * 60 * 60 * 1000 * 365,
+    month: 24 * 60 * 60 * 1000 * 365/12,
+    day: 24 * 60 * 60 * 1000,
+    hour: 60 * 60 * 1000,
+    minute: 60 * 1000,
+    second: 1000
+  };
+
+  const rtf = new Intl.RelativeTimeFormat('en', { numeric: "auto", style: 'long' });
+  const elapsed = (timestamp * 1000) - current;
```

Comment:
We should check if `current >= timestamp` and just return `"now"` in that case. We can then also drop the `Math.abs` usage.

Reply:
Ok did rewrite this, the only thing we have to maintain is when calling `rtf.format` we have to pass it a negative number which I do by multiplying by `-1`, else we get `in 1 day` or similar output
While before doing `timestamp - current` we obtained the elapsed time with the right sign